### PR TITLE
Fix a bug in the previous gelu implementation

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
Fix a bug in the previous gelu implementation
That was really a silly bug. I forget to set correct output ptr.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by pranavsharma** [comment]
Is this implementation faster than the previous one which was based on FastGelu? cc @tianleiwu

**Comment by tianleiwu** [comment]
Here is perf test on my machine (i7-8700 CPU @ 3.20GHz, 12 logical cores, 6 physical cores):
|              |                  |                 |                 |             |             | 
|--------------|------------------|-----------------|-----------------|-------------|-------------| 
| input_length | intra_op_threads | OMP_NUM_THREADS | OMP_WAIT_POLICY | Avg_Latency (master)| Avg_Latency (this PR)| 
| 98304        | 0                |                 |                 | 0.095       | 0.098       | 
| 1572864      | 0                |                 |                 | 1.883       | 2.06        | 
| 98304        | 0                | 12              | ACTIVE          | 0.092       | 0.099       | 
| 1572864      | 0                | 12              | ACTIVE          | 1.956       | 2.648       | 
| 98304        | 0                | 6               | PASSIVE         | 0.09        | 0.088       | 
| 1572864      | 0                | 6               | PASSIVE         | 1.286       | 2.088       | 

Master version is faster.

**Comment by snnn** [comment]
@tianleiwu: Yes, in my local test, it shows the similar thing, when input length = 1572864, the master version is about 20% faster than mine, but when input length = 10000, the master version is 50% slower.

File: onnxruntime/test/contrib_ops/activation_op_test.cc [hunk 1]
```diff
@@ -4,71 +4,79 @@
...
+}  // namespace test
 
-std::vector<float> input_values = {
-    -1.0f, 0, 1.0f,                                              // normal input values for activation
-    100.0f, -100.0f, 1000.0f, -1000.0f,                          // input values that leads to exp() overflow
-    FLT_MIN, FLT_MIN / 10, -FLT_MIN / 10,                        // min, denorm, -denorm
-    FLT_MAX, -FLT_MAX, std::numeric_limits<float>::infinity()};  // max, -max, inf
+class ActivationContribOpTest : public ::testing::Test {
+ protected:
+  std::vector<std::vector<float>> input_values;  // max, -max, inf
```
Comment (yufenglee) [inline onnxruntime/test/contrib_ops/activation_op_test.cc:43]: >  // max, -max, inf [](start = 47, length = 19)

Add check for those special cases

**Comment by yufenglee** [comment]
The typical tensor size is Batch*128*768. If batch == 1, it is 

> @tianleiwu: Yes, in my local test, it shows the similar thing, when input length = 1572864, the master version is about 20% faster than mine, but when input length = 10000, the master version is 50% slower.

@snnn, the typical input size is batch*128*768 = batch * 98304.

**Comment by tianleiwu** [comment]
My previous test is one model with a single Gelu op. I also have results on Bert base model with Gelu using openmp. Note that BiasGelu fusion is disabled so that the optimized model has Gelu operator. There are two set of settings: one set is default (no OMP setting), another with OMP setting. sequence_length=128 has input length 1572864 for Gelu, and sequence_length=16 has input length 98304.

Among 10 settings below, master is better on majority, and this PR wins only 1.

| sequence_length | OMP_NUM_THREADS | OMP_WAIT_POLICY | avg latency (master) | avg latency (this PR) | 
|-----------------|-----------------|-----------------|------------------|-------------------| 
| 8               |                 |                 | 28.17            | 28.25             | 
| 16              |                 |                 | 32.16            | 32.74             | 
| 32              |                 |                 | 42.37            | 42.86             | 
| 64              |                 |                 | 65.39            | 66.45             | 
| 128             |                 |                 | 110.37           | 111.70            | 
|                 |                 |                 |                  |                   | 
| 8               | 12              | PASSIVE         | 31.27            | 31.31             | 
| 16              | 12              | PASSIVE         | 32.59            | 32.87             | 
| 32              | 12              | PASSIVE         | 41.50            | **40.92**             | 
| 64              | 12              | PASSIVE         | 58.23            | 60.32             | 
| 128             | 12              | PASSIVE         | 101.49           | 102.51            |

**Pull request closed without merging**